src/components/TodoForm/TodoForm.tsx

```diff
@@ -0,0 +1,219 @@
+// #region imports
+import classNames from 'classnames';
+import React, { useEffect, useRef, useState } from 'react';
+import { Todo } from '../../types/Todo';
+import { getAllUsers, getUserById } from '../../services/user';
+// #endregion
+
+type Props = {
+  onSubmit: (todo: Todo) => void;
+  onReset?: () => void;
+  todo?: Todo;
+};
+
+export const TodoForm: React.FC<Props> = ({
+  onSubmit,
+  todo,
+  onReset = () => {},
+}) => {
+  const titleField = useRef<HTMLInputElement>(null);
+
+  //const [newTitle, setNewtitle] = useState('blablabla');
+  //const [isBodyShown, setIsBodyShown] = useState(false);
+
+  useEffect(() => {
+    if (titleField.current && todo) {
+      titleField.current.focus();
+    }
+  }, [todo]);
+  //[post?.id]
+
+  // #region state
+  const [title, setTitle] = useState(todo?.title || '');
+  const [hasTitleError, setHasTitleError] = useState(false);
+
+  const [userId, setUserId] = useState(todo?.userId || 0);
+  const [hasUserIdError, setHasUserIdError] = useState(false);
+
+  const [body, setBody] = useState(todo?.body || '');
+  const [bodyErrorMessage, setBodyErrorMessage] = useState('');
+  // #endregion
+  // #region handlers
+  const handleTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value);
+    setHasTitleError(false);
+  };
+
+  const handleUserIdChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setUserId(+event.target.value);
+    setHasUserIdError(false);
+  };
+
+  const handleBodyChange = (event: React.ChangeEvent<HTMLTextAreaElement>) => {
+    setBody(event.target.value);
+    setBodyErrorMessage('');
+  };
+
+  // #endregion
+  // #region reset
+  const reset = () => {
+    setTitle('');
+    setUserId(0);
+    setBody('');
+
+    setHasTitleError(false);
+    setHasUserIdError(false);
+    setBodyErrorMessage('');
+
+    onReset();
+  };
+
+  // #endregion
+  // #region submit
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    setHasTitleError(!title);
+    setHasUserIdError(!userId);
+
+    if (!body) {
+      setBodyErrorMessage('Please enter some text');
+    } else if (body.length < 5) {
+      setBodyErrorMessage('Body should have at least 5 chars');
+    }
+
+    if (!title || !userId || body.length < 5) {
```

Comment:
The `handleSubmit` function should prevent form submission if there are validation errors. Ensure that the function returns early if any validation error is present.